docs/stories/WebTools.md

```diff
@@ -0,0 +1,53 @@
+# Web Tools
+
+## Configure web application directory in WTP configurations for web application projects
+
+### Motivation
+The `ide-integration.md` document in Gradle core specifies a story ["Expose web application directory"](...). The goal is to define an initial end-to-end framework for Buidship/WTP integration, and to
+communicate the location of the web application directory to WTP.
+
+### Estimate
+
+- ? day(s)
+
+### Implementation
+- Create branch in buildship repository `wtp-integration`.
+- Define `org.eclipse.buildship.wtp` plugin in `wtp-integration`.
+- Define `org.eclipse.buildship.wtp.test` plugin in `wtp-integration`.
+- Wait for corresponding story in TAPI to be completed.
+- Upgrade Tooling Commons (if necessary).
+- Define configuration class in `org.eclipse.buildship.wtp`, either configurator class, or a similar class to `DefaultWorkspaceGradleOperations`.
+    - Retrieve Web application model from TAPI
+    - If a project is a web application project (model will be `null` if project doesn't apply `war` plugin)
+        - Add Java and Dynamic Web Facets to project, which will generate component file
+        - Configure web app dir location in component file
+    - Otherwise, do nothing
+- Invoke WTP configuration from Buildship core
+
+### Test cases
+- Configuration only invoked on web application projects
+- Java facet is added to project
+- Dynamic Web Facet is added to project
+- If Java Facet already on project but has incorrect version, version is updated to correct version based on source Java version
+- If Dynamic Web Facet already on project but has incorrect version, version is updated to correct version
+- Default Java Facet version is x.
+- Default Dynamic Web Facet version is x.
+- If Java Facet exists on project with correct version, Facet is not affected.
+- If Dynamic Web Facet exists on project with correct version, Facet is not affected.
+- Custom web application directory location is communicated successfully to component file, file exists.
+- Default web application directory location is communicated successfully to component file, file exists.
+- Custom web application directory location is communicated successfuly to component file, file does not exist.
+- Default web application directory location is communicated successfully to component file, file does not exist.
+
```

Comment:
gradle-controlled paths = list of paths inferred from build.gradle, that were added as wb-resource entries during the previous configuration round

there's no API to fine tune wb-resource entries and add extra metadata. Writing to .component from outside the API is a no-go, would lead to synchronization issues. There's also a very good chance every change made to custom wb-resource attributes would be deleted when the user interacts with the Deployment Assembly page.